
This patch is needed if you want to recompile the SWIG wrappers.

                                             -- Gustavo Niemeyer


Index: Source/Modules1.1/lang.cxx
===================================================================
--- Source/Modules1.1/lang.cxx
+++ Source/Modules1.1/lang.cxx	2002-11-21 00:45:46.000000000 -0200
@@ -1457,13 +1457,19 @@
 	}
       }
     } else {
-      if (name && (Cmp(name,ClassName)) && !(Getattr(n,"template"))) {
+      String *namelast = Swig_scopename_last(name);
+      String *classlast = Swig_scopename_last(ClassName);
+      if (name && (Cmp(namelast,classlast)) && !(Getattr(n,"template"))) {
 	Printf(stdout,"name = '%s', ClassName='%s'\n", name, ClassName);
 	Swig_warning(WARN_LANG_RETURN_TYPE, input_file,line_number,"Function %s must have a return type.\n", 
 		     name);
 	Swig_restore(&n);
+	Delete(namelast);
+	Delete(classlast);
 	return SWIG_NOWRAP;
       }
+      Delete(namelast);
+      Delete(classlast);
       constructorHandler(n);
     }
   }
Index: Source/Swig/symbol.c
===================================================================
--- Source/Swig/symbol.c
+++ Source/Swig/symbol.c	2002-11-21 00:46:03.000000000 -0200
@@ -762,11 +762,16 @@
     } else {
       String *prefix = Swig_scopename_prefix(name);
       if (prefix) {
-	s = symbol_lookup_qualified(name,hsym,0,0);
+	Node *pn = Swig_symbol_clookup(prefix,hsym);
+	if (pn && Strcmp(nodeType(pn),"namespace") == 0) {
+	  s = symbol_lookup_qualified(name,hsym,0,0);
+	}
 	Delete(prefix);
+	/*
 	if (!s) {
 	  return 0;
 	}
+	*/
       }
     }
   }
